nft_ingester/tests/api_tests.rs

```diff
@@ -4109,6 +4109,406 @@ mod tests {
+            )
+            .unwrap();
+        token_accounts_processor
+            .transform_and_save_token_account(
+                &mut batch_storage,
+                fungible_token_account2,
+                &token_account4,
+            )
+            .unwrap();
+        token_accounts_processor
+            .transform_and_save_mint_account(&mut batch_storage, &mint1, &Default::default())
+            .unwrap();
+        token_accounts_processor
+            .transform_and_save_mint_account(&mut batch_storage, &mint2, &Default::default())
+            .unwrap();
+        token_accounts_processor
+            .transform_and_save_mint_account(&mut batch_storage, &mint3, &Default::default())
+            .unwrap();
+        token_accounts_processor
+            .transform_and_save_mint_account(&mut batch_storage, &mint4, &Default::default())
+            .unwrap();
+        batch_storage.flush().unwrap();
+
+        let synchronizer = Arc::new(synchronizer);
+
+        for asset_type in ASSET_TYPES {
+            let synchronizer = synchronizer.clone();
+            let _ = match asset_type {
+                AssetType::Fungible => {
+                    synchronizer
+                        .synchronize_fungible_asset_indexes(CancellationToken::new(), 0)
+                        .await
+                },
+                AssetType::NonFungible => {
+                    synchronizer.synchronize_nft_asset_indexes(CancellationToken::new(), 0).await
+                },
+            };
+        }
+
+        let api = create_api(&env, None);
+
+        let mut pubkeys_sorted = [nft_mint1, nft_mint2, fungible_token_mint1, fungible_token_mint2];
+        pubkeys_sorted.sort();
+        pubkeys_sorted.reverse();
+        let payload = SearchAssets {
+            limit: Some(2),
+            owner_address: Some(owner.to_string()),
+            options: SearchAssetsOptions {
+                show_zero_balance: true,
+                show_unverified_collections: true,
+                ..Default::default()
+            },
+            token_type: Some(TokenType::All),
+            ..Default::default()
+        };
+        let res = api.search_assets(payload).await.unwrap();
+        let res: AssetList = serde_json::from_value(res).unwrap();
+
+        // the default direction of sorting is descending.
+        assert_eq!(
+            [res.items[0].id.clone(), res.items[1].id.clone()],
+            [pubkeys_sorted[0].to_string(), pubkeys_sorted[1].to_string()],
+            "Should return the items in descending order"
+        );
+
+        let cursor = res.cursor.unwrap();
+
+        let payload = SearchAssets {
+            limit: Some(2),
+            owner_address: Some(owner.to_string()),
+            options: SearchAssetsOptions {
+                show_zero_balance: true,
+                show_unverified_collections: true,
+                ..Default::default()
+            },
+            token_type: Some(TokenType::All),
```

Comment:
I would add some tests with search on TokenType NonFungible to cover another branch of the query builder.

Reply:
the pagination worked correctly for NFTs; it worked incorrectly for fungible tokens. so there's not much value in testing NFTs specifically